portal-impl/src/com/liferay/portal/verify/VerifyAuditedModel.java

```diff
@@ -214,142 +174,169 @@ protected String getUserName(Connection con, long userId) throws Exception {
 	}
 
 	protected void verifyAuditedModel(
-			Connection con, PreparedStatement ps, String tableName,
-			long primKey, Object[] auditedModelArray, boolean updateDates)
+			VerifiableAuditedModel verifiableAuditedModel)
 		throws Exception {
 
-		try {
-			long companyId = (Long)auditedModelArray[0];
+		try (LoggingTimer loggingTimer = new LoggingTimer(
+				verifiableAuditedModel.getTableName());
+			Connection con = DataAccess.getUpgradeOptimizedConnection()) {
 
-			if (auditedModelArray[2] == null) {
-				auditedModelArray = getDefaultUserArray(con, companyId);
+			String relatedModelName =
+				verifiableAuditedModel.getRelatedModelName();
 
-				if (auditedModelArray == null) {
-					return;
-				}
-			}
+			if (relatedModelName != null) {
+				_bulkVerifyFromRelatedModel(
+					con, verifiableAuditedModel, "userId");
 
-			long userId = (Long)auditedModelArray[1];
-			String userName = (String)auditedModelArray[2];
-			Timestamp createDate = (Timestamp)auditedModelArray[3];
-			Timestamp modifiedDate = (Timestamp)auditedModelArray[4];
+				if (verifiableAuditedModel.isUpdateDates()) {
+					_bulkVerifyFromRelatedModel(
+						con, verifiableAuditedModel, "createDate");
 
-			ps.setLong(1, companyId);
-			ps.setLong(2, userId);
-			ps.setString(3, userName);
+					_bulkVerifyFromRelatedModel(
+						con, verifiableAuditedModel, "modifiedDate");
+				}
 
-			if (updateDates) {
-				ps.setTimestamp(4, createDate);
-				ps.setTimestamp(5, modifiedDate);
-				ps.setLong(6, primKey);
-			}
-			else {
-				ps.setLong(4, primKey);
-			}
+				DBInspector dbInspector = new DBInspector(con);
 
-			ps.addBatch();
-		}
-		catch (Exception e) {
-			if (_log.isWarnEnabled()) {
-				_log.warn("Unable to verify model " + tableName, e);
+				if (dbInspector.hasColumn(relatedModelName, "userName")) {
+					_bulkVerifyFromRelatedModel(
+						con, verifiableAuditedModel, "userName");
+				}
 			}
+
+			_bulkVerifyRegularUsers(con, verifiableAuditedModel);
+
+			_bulkVerifyUnresolvedUsers(con, verifiableAuditedModel);
 		}
 	}
 
-	protected void verifyAuditedModel(
-			VerifiableAuditedModel verifiableAuditedModel)
+	private void _bulkVerifyFromRelatedModel(
+			Connection con, VerifiableAuditedModel verifiableAuditedModel,
+			String columnName)
 		throws Exception {
 
-		try (LoggingTimer loggingTimer = new LoggingTimer(
-				verifiableAuditedModel.getTableName())) {
+		String sql = StringBundler.concat(
+			"update ", verifiableAuditedModel.getTableName(), " set ",
+			columnName, " = (select ", columnName, " from ",
+			verifiableAuditedModel.getRelatedModelName(), " where ",
+			verifiableAuditedModel.getTableName(), StringPool.PERIOD,
+			verifiableAuditedModel.getJoinByTableName(), " = ",
+			verifiableAuditedModel.getRelatedModelName(), StringPool.PERIOD,
+			verifiableAuditedModel.getRelatedPKColumnName(),
+			") where userName is null");
```

Comment:
@holatuwol using "is null" comparison can be dangerous sometimes because we need to compare with "" (empty) too. Anyway we were using "is null" before your changes